WordPress/gutenberg
# Widgets: Add Welcome Guide

## Description

Requires https://github.com/WordPress/gutenberg/pull/31926.
Implements part of https://github.com/WordPress/gutenberg/issues/30887.

Adds `WelcomeGuide` to the widgets screen. This is mostly copied from `WelcomeGuide` in `edit-post`.

We show the "new user" copy when all widgets are block widgets, because when this feature is merged to Core, we will update `wp_install_defaults` ([here](https://github.com/WordPress/wordpress-develop/blob/master/src/wp-admin/includes/upgrade.php#L404)) to set up a sidebar containing block widgets in new WordPress installations. We can't do this in the Gutenberg plugin, though.

## How has this been tested?

1. Go to Appearance → Widgets. You should see the welcome guide.
2. Dismiss the guide and refresh the page. You should not see the welcome guide.
3. Replace every Legacy Widget with a block.
4. Clear local storage e.g. by running `localStorage.clear()`.
5. Refresh the page. You should see the welcome guide with different copy.

## Screenshots

## Checklist:
